core/src/BibleEngine.class.ts

```diff
@@ -777,16 +780,21 @@ export class BibleEngine {
                     versionId: state.book.versionId,
                     phraseNum: state.currentPhraseNum
                 };
+
                 const phraseId = generatePhraseId(phraseRef);
+                if (phraseId in state.alreadyUsedPhraseIds) {
+                    throw new Error('Duplicate phrase id found.');
+                }
+                state.alreadyUsedPhraseIds[phraseId] = content;
                 if (!firstPhraseId) firstPhraseId = phraseId;
                 lastPhraseId = phraseId;
 
                 if (state.columnModifierState.quoteWho)
                     content.quoteWho = state.columnModifierState.quoteWho;
                 if (state.columnModifierState.person)
                     content.person = state.columnModifierState.person;
-
-                state.phraseStack.push(new BiblePhrase(content, phraseRef, state.modifierState));
+                const phrase = new BiblePhrase(content, phraseRef, state.modifierState);
```

Comment:
i guess this is a style issue? my thinking would be to not create a new constant when not needed.

apart from that - as mentioned in slack - i think we need to pass a cloned state object here (my bug).